acra/configuring-maintaining/general-configuration/acra-keys/generate.md

```diff
@@ -0,0 +1,194 @@
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `https://example.com:8200`) for connecting to HashiCorp Vault.
+  If not specified, `ACRA_MASTER_KEY` environment variable will be used.
+
+* `--vault_secrets_path=<kv-path>`
+
+  Path to KV Secrets directory in Vault used to store `ACRA_MASTER_KEY`.
+  Default is `secret/`.
+
+* `--vault_tls_ca_path=<path>`
+
+  Path to CA certificate bundle to use for HashiCorp Vault certificate validation.
+
+  If not specified, use root certificates configured in system.
+
+* `--vault_tls_client_cert=<path>`
+
+  Path to client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_client_key=<path>`
+
+  Path to the private key of the client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_transport_enable={true|false}`
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+
+## Generating keys
+
+{{< hint warning >}}
+**Note:**
+Back up all your keys before calling `acra-keys generate` with any flag described below.
+Some keys can be rotated and the previous keys will be retained,
+while some keys will be overwritten.
+{{< /hint >}}
+
+### Acra master key
+
+* `--master_key_path=<path>`
+
+  Generate a new random master key and write it to the specified file.
+
+  All Acra components will expect this key in `ACRA_MASTER_KEY` environment variable.
+
+{{< hint info >}}
+**Note:**
+When you generate a new key (or keypair), the existing key is rotated.
+That is, the new key becomes the default for encryption in [Transparent proxy mode]({{< ref "/acra/configuring-maintaining/general-configuration/acra-server.md#transparent-proxy-mode-INVALID" >}})
+while the previous key is archived and used only for decryption.
+{{< /hint >}}
+
+### Transport keys
+
+* `--acraconnector_transport_key`
+
+  Generate a new transport keypair for AcraConnector.
+
+* `--acraserver_transport_key`
+
+  Generate a new transport keypair for AcraServer.
+
+* `--acratranslator_transport_key`
+
+  Generate a new transport keypair for AcraTranslator.
+
+
+### AcraServer & AcraTranslator keys
+
+* `--search_hmac_symmetric_key`
+
+  Generate a new key for HMAC calculation and searchable encryption.
+
+* `--audit_log_symmetric_key`
+
+  Generate a new key symmetric key for log integrity checks.
+
+* `--poison_record_keys`
+
+  Generate a new keypair and symmetric key for poison records.
+
+* `--client_storage_symmetric_key`
+
+  Generate a new symmetric key for data encryption/decryption with AcraBlock.
+
+* `--client_storage_key`
+
+  Generate keypair for data encryption/decryption (for a client).
+
+
+### AcraServer & AcraAuthManager keys
+
+* `--acrawebconfig_symmetric_key`
+
+  Generate a new symmetric key for encrypting AcraWebconfig's basic authentication credentials.
+
+
+### Zone keys
+
+* `--zone`
+
+  Generate new Acra storage zone.
```

Comment:
it is not obvious flag description that pushed me to read source code to understand what actually it doing)
`--zone` generates new Zone keypair (without symmetric key). and `--zone_storage_key` rotates existing by generating new and changing current to read-only state, for decryption purposes only.